src/tools/sc-hsm-tool.c

```diff
@@ -803,10 +821,9 @@ static void ask_for_password(char **pwd, int *pwdlen)
 
 
 
-static int generate_pwd_shares(sc_card_t *card, char **pwd, int *pwdlen, int password_shares_threshold, unsigned int password_shares_total)
+static int generate_pwd_shares(sc_card_t *card, char **pwd, int *pwdlen, int password_shares_threshold, int password_shares_total)
```

Comment:
Because -1 is the default value of any option and I have the habit to do semantical checks close to were the actual processing code is.
Sure, one could define 0 as the default value meaning "Option not defined". I could as well say: "Hey 34563621 is the default value". Why is it important to make it an unsigned int rather than an int ? The range we expect is for these parameter is somewhere between 2 and 10, no real range issue here, nothing to save by converting to unsigned and no benefit in parameter checking - other than having to rewrite working code.